### ISSUE TITLE
[rllib] unroll_id is not repeatable once upgraded to 0.8.7

### ISSUE THREAD

**Issue by iamhatesz** [ray-project/ray#10156]
### What is the problem?

*Ray version and other system information (Python version, TensorFlow version, OS):* 0.8.7, Python 3.7.7, Ubuntu 20.04

### Reproduction (REQUIRED)

Rollouts collected with the same seed are now varying on `unroll_id` field. The same script worked for me before upgrading to 0.8.7 from 0.8.6. If we skip assertion on `unroll_id`, then the test passes.

```python
def make_trainer(seed: int):
    return PPOTrainer(
        config={
            "env": "CartPole-v1",
            "framework": "torch",
            "num_workers": 2,
            "rollout_fragment_length": _NUM_STEPS,
            "train_batch_size": _NUM_STEPS,
            "sgd_minibatch_size": _NUM_STEPS,
            "num_sgd_iter": 1,
            "seed": seed,
        }
    )


_SEEDS = [0, 1337]
_NUM_STEPS = 50


class TestRaySeed(unittest.TestCase):
    def setUp(self) -> None:
        ray.init(local_mode=True, ignore_reinit_error=True)

    def tearDown(self) -> None:
        ray.shutdown()

    def test_samples_collected_are_equal(self):
        for seed in _SEEDS:
            with self.subTest(seed=seed):
                trainer1 = make_trainer(seed)
                rollout1 = trainer1.workers.foreach_worker(lambda ev: ev.sample())

                trainer2 = make_trainer(seed)
                rollout2 = trainer2.workers.foreach_worker(lambda ev: ev.sample())

                for sub_rollout1, sub_rollout2 in zip(rollout1, rollout2):
                    for (name, value1), (_, value2) in zip(
                        sub_rollout1.items(), sub_rollout2.items()
                    ):
                        # Uncomment these 2 lines to make it work
                        # if name == "unroll_id":
                        #     continue
                        np.testing.assert_equal(value1, value2, name)
```

- [x] I have verified my script runs in a clean environment and reproduces the issue.
- [ ] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/latest/installation.html).

**Comment by ericl** [comment]
I think this is because local_mode=True. If you disable that, then unroll id should be isolated per process as usual. We did some bug fixes in 0.8.7 that made the unroll id counter a per-process variable to resolve an issue where it was duplicated.
